doc/pages/architecture/090-database.md

```diff
@@ -0,0 +1,219 @@
+# Astarte Database
+
+Astarte might need to store a massive ammount of data, to accomplish this task a [Cassandra](http://cassandra.apache.org/) database is used. Cassandra offers scalability and high availability with [good performances](https://www.datastax.com/apache-cassandra-leads-nosql-benchmark).
+Cassandra offers linear scalability and can span from really small clusters to thousands of nodes, last but not least it provides a reliable storage for all devices data.
+[ScyllaDB](https://www.scylladb.com/) >= 2.5 is also supported as a drop-in replacement when a [performances boost](https://www.scylladb.com/product/benchmarks/) is needed.
+
+Cassandra is also the ideal storage for large-scale data processing with [Apache Spark](http://spark.apache.org/).
+
+Astarte is multi-tenant by design, each tenant has its own realm. Each realm has its own Cassandra keyspace, that can be tuned according to realm specific needs (e.g. each realm might have a different replication level). For this reason here realm and keyspace can be used as synonyms, except for the `astarte` keyspace.
+
+## Schema and Keyspace Creation
+
+Astarte automatically takes care of keyspaces and tables creation (those tasks are performed by `astarte_housekeeping`), here are documented for completeness.
+
+### Astarte Keyspace
+
+Astarte needs an `astarte` keyspace to store its own data.
+
+`astarte` keyspace and tables are created with following [CQL](https://docs.datastax.com/en/cql/3.3/index.html) statements:
+
+```
+CREATE KEYSPACE astarte
+  WITH replication = {'class': 'SimpleStrategy', 'replication_factor': <replication factor>}  AND
+    durable_writes = true;
+```
+
+```
+CREATE TABLE astarte.realms (
+  realm_name varchar,
+  replication_factor int,
+
+  PRIMARY KEY (realm_name)
+);
+```
+
+```
+CREATE TABLE astarte.astarte_schema (
+  config_key varchar,
+  config_value varchar,
+  PRIMARY KEY (config_key)
+);
+```
```

Comment:
If we're merging this into 0.11 (assume we are), ensure the statements are consistent with what we have in 0.11